discord/http.py

```diff
@@ -256,6 +259,13 @@ def handle_message_parameters(
         }
         payload.update(channel_payload)
 
+    if poll is not MISSING:
+        if len(poll) == 0 or len(poll) > 10:
+            raise ValueError('Poll must contain between 1 and 10 answers')
+        if poll._hours_duration < 1 or poll._hours_duration > 168:
+            raise ValueError('Polls duration must be between 1 hour and 7 days')
+        payload['poll'] = poll._to_dict()  # type: ignore
```

Comment:
You need to check poll isn't None prior to this.

Reply:
Fixed in latest commits